### ISSUE TITLE
Fix namespace misuse

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#706]
Cxbx-Reloaded currently misuses namespace declarations: in EmuXTL.h, source code files are #include'd from within an XTL namespace, resulting in all included symbols being declared in the XTL namespace. Any use of the same include files outside of the XTL namespace intrudes another declaration of the same time symbols, causing duplicate symbols and many occurrences of inconvenient and hard to fix compile errors.

For this issue:
* Investigate the issue
* Describe a better approach
* Discuss with the team
* Reach consensus
* Implement the chosen approach

(Note: All includes from openxdk are also included within a namespace (xboxkrnl). Since these includes are imported from an external repository, we don't want to change their namespace use internally. So these we should keep including from within the xboxkrnl namespace.)

**Comment by RadWolfie** [comment]
Yeah, I have seen a lot and have to deal with certain files which does not have XTL namespace. While other files are including these with XTL namespace wrapper... Definitely a pain in the neck.

**Comment by LukeUsher** [comment]
The XTL namespace is designed for Xbox symbols, to separate Xbox data types and code from native code, and to prevent naming conflicts when Xbox functions have the same names as Windows equivalent.  Patches are in the XTL namespace for this same reason.

The problem is this standard hasn't been enforced over time, and new functions have been added all over the place with no real structure.

**Comment by x1nixmzeng** [comment]
I've just noticed how insane the XTL namespace is!

There are headers such as `EmuXInput.h` which rely on the _header include order_ to work correctly, as it's only wrapped in a namespace when including `EmuXTL.h` first!

It obviously compiles, but it's confusing the intellisense!

We can keep the XTL namespace but it's usage throughout must be consistent. One pattern I've found in C# is nesting namespaces such as `Constants` or `NativeMethods` which is crystal clear.
